src/neighborsearch.jl

```diff
@@ -27,12 +27,12 @@ specify a `mask` for all indices of the domain.
 function search end
 
 """
-    BoundedNeighborSearchMethod
+    FilteredNeighborSearchMethod
 
 A method for searching neighbors with the property that the number of neighbors
 is bounded above by a known constant (e.g. k-nearest neighbors).
 """
-abstract type BoundedNeighborSearchMethod <: NeighborSearchMethod end
+abstract type FilteredNeighborSearchMethod <: NeighborSearchMethod end
```

Comment:
I think we want to stick to BoundedNeighborSearchMethod here, it is just an abstract type that informs when a search method is guaranteed to return a bounded number of indices.